make.js

```diff
@@ -49,36 +53,53 @@ if (options.build) {
 }
 
 if (options.test) {
+    const gitkeepName = '.gitkeep';
+    const junitPath = path.join(testResultsPath, 'junit');
+    const coveragePath = path.join(testResultsPath, 'coverage');
+
     console.log('Testing shared npm packages');
     util.cd('common-npm-packages');
     const suite = options.suite || defaultTestSuite;
     let testsFailed = false;
+    util.cleanFolder(testResultsPath, [gitkeepName]);
 
-    fs.readdirSync('./', { encoding: 'utf-8' }).forEach(child => {
+    const startPath = process.cwd();
+    fs.readdirSync(startPath, { encoding: 'utf-8' }).forEach(child => {
         if (fs.statSync(child).isDirectory() && !ignoredFolders.includes(child)) {
             printLabel(child);
+            const buildPath = path.join(startPath, child, '_build');
 
-            if (fs.existsSync(path.join('./', child, '_build'))) {
-                util.cd(path.join(child, '_build'));
+            if (fs.existsSync(buildPath)) {
+                const testPath = path.join(buildPath, 'Tests', `${suite}.js`);
 
-                if (fs.existsSync(path.join('./', 'Tests', `${suite}.js`))) {
+                if (fs.existsSync(path.join(testPath))) {
                     try {
-                        util.run(`mocha Tests/${suite}.js`, true);
+                        const suitName = `${child}-suite`;
+                        const mochaOptions = util.createMochaOptions(mochaReporterPath, junitPath, suitName);
+
+                        util.run(`nyc --all --src ${buildPath} --report-dir ${coveragePath} mocha ${mochaOptions} ${testPath}`, true);
+                        util.renameFile(coveragePath, coverageBaseNameJson, `${child}-coverage.json`);
                     } catch (err) {
                         testsFailed = true;
-                    } finally {
-                        util.cd('../..');
                     }
                 } else {
                     console.log('No tests found for the package');
-                    util.cd('../..');
                 }
             } else {
                 throw new Error('Package has not been built');
             }
         }
     });
+
     if (testsFailed) {
         throw new Error('Tests failed!');
     }
-}
+
+    try {
+        util.rm(path.join(coveragePath, summaryBaseName));
+        util.run(`nyc merge ${coveragePath} ${path.join(testResultsPath, 'merged-coverage.json')}`, true);
+        util.run(`nyc report -t ${testResultsPath} --report-dir test-results --reporter=cobertura`, true);
```

Comment:
It exists const `testResultsPath` with this folder `test-results`. Maybe it's needed to create const with value `test-results`?

Reply:
Goot point, thanks